### ISSUE TITLE
context.__styled-components__ as a function is deprecated

### ISSUE THREAD

**Issue by benjamingeorge** [styled-components/styled-components#1191]
I recently upgraded to React 16 and Styled-Components 2.2.0 inside an ejected version of Create-React-App

I'm getting this error in the console and was wondering if anyone else has come across it and could shed some light on the cause of it.  I've googled around and can't seem anything references for it.

`Warning: Usage of `context.__styled-components__` as a function is deprecated. It will be replaced with the object on `.context.__styled-components__next__` in a future version.`

**Comment by philpl** [comment]
Do you have any custom code around theming that is not using the high-level API? We recently changed how the styled-components theming context works, so we put this deprecation message in

**Comment by genox** [comment]
I'm relatively new to styled components and get this notification. I wonder if you can exemplify the specific changes that this implies.

For example, I'm using the module styled-mixin to achieve more sophisticated styling, which looks like this in a component:

``
const Wrapper = block.sectionBlock(styled.section`
  background: ${theme.palette.paper};
`)

const TitleH01 = typo.h01(styled.h1``)
const BodyP01 = typo.p01(styled.p``)

const TitleH02 = typo.h02(styled.h2``)
const BodyP02 = typo.p02(styled.p``)

const Block = block.pBlock(styled.div``)
``

So, in a nutshell: What do I need to consider to write my styling to be future proof in regards to the changes that the message announces?

**Comment by benjamingeorge** [comment]
Upgrading other dependencies today such as grid-styled, styled-system solved the issue.  I guess those library authors fixed the issue in their codebase.

**Comment by philpl** [comment]
@benjamingeorge Thanks for following this up for us :+1:
